src/app/users/editServices.js

```diff
@@ -18,6 +19,18 @@ const renderEditServicePage = async (req, res, model) => {
   const userDetails = await getUserDetails(req);
   const isManage = isUserManagement(req);
 
+  // Save the session explicitly as both the get and post functions attempt to modify
+  // the session.
+  req.session.save((err) => {
+    if (err) {
+      // Any error saving to session should hopefully be temporary. Assuming this, we log the error
+      // and just display an error message saying to try again.
+      logger.error("An error occurred when saving to the session", err);
+      model.validationMessages.roles =
```

Comment:
Before assigning model.validationMessages.roles you should ensure model.validationMessages exists to avoid runtime exceptions